ndcube/ndcube.py

```diff
@@ -615,7 +620,33 @@ def _bounding_box_to_points(self, lower_corner_values, upper_corner_values, wcs)
         """
         Convert two corners of a bounding box to the points of all corners.
         """
-        return tuple(itertools.product(*zip(lower_corner_values, upper_corner_values)))
+        corners = np.array(tuple(itertools.product(*zip(lower_corner_values, upper_corner_values))),
+                           dtype=object)
+        if hasattr(wcs, "mapping"):
```

Comment:
This code is to catch circumstances where world axes share pixel axes.  The first line of this method would create 2^n corners where `n` is the number of world axes.  However, only 2 of these corners are valid as they all share the same axis.  This will cause `world_to_pixel` to error.

This solution however, is specific to a WCS implementation.  It should be refactored to detect the situation based on the `axis_correlation_matrix`.  Exactly how to do this requires more thought.